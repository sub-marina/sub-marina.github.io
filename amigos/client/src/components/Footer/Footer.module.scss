@import '../../scss/base/base';

.footer {
  position: relative;
  width: 100%;
  padding: base-gutter(5) 0 base-gutter(3);
  background-color: $black-color;

  &::before {
    content: '';
    width: 100%;
    height: 3px;
    position: absolute;
    top: 0;
    left: 0;
    background-color: $orange-color;
    opacity: 0.3;
  }

  .footerWrap {
    height: 160px;
    @include flex(space-between);
    flex-direction: column;

    & > div:first-child {
      @include flex();
      flex-direction: column;
    }
  }

  .logo {
    width: 110px;
    margin-bottom: 15px;

    img {
      width: 100%;
    }
  }

  .footerLink {
    text-decoration: none;
    text-transform: uppercase;
    color: #ffffff;
    line-height: 24px;
  }

  .copyright {
    font-size: 12px;
    color: #ffffff;
  }
}

@include resp-md {
  .footer {
    .footerWrap {
      height: auto;
      flex-direction: row;
      justify-content: space-between;

      & > div:first-child {
        width: 415px;
        flex-direction: row;
        justify-content: space-between;
        align-items: unset;
      }
    }

    .logo {
      margin-bottom: 0;
      width: 90px;
    }
  }
}