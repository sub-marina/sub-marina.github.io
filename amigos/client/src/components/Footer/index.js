import React from 'react';
import styles from './Footer.module.scss';
import { Link } from 'react-router-dom';
import Button from '../Button';

const scrollTop = () => window.scrollTo({top: 0, behavior: 'smooth'});

const Footer = () => (
    <div className={styles.footer}>
        <Button action={scrollTop} classList='btn footer__btnUp' children='&#x2303;' />
        <div className={`container ${styles.footerWrap}`}>
            <div>
                <div className={styles.logo}>
                    <Link to='/'><img src='/assets/logo.png' alt='' /></Link>
                </div>
                <Link to='/about' className={styles.footerLink}>О нас</Link>
                <Link to='/questions' className={styles.footerLink}>Вопросы и ответы</Link>
                <Link to='/contacts' className={styles.footerLink}>Контакты</Link>
            </div>
            <div className={styles.copyright}>
                &copy; 2021 Amigos. Все права защищены.
            </div>
        </div>
    </div>
);

export default Footer;