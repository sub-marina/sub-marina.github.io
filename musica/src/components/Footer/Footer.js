import React from 'react';
import Button from '../Button/Button';

const scrollTop = () => window.scrollTo({top: 0, behavior: 'smooth'});

const Footer = () => {
    return (
        <div className='main-footer'>
            <div className='main-footer__upper-line' />
            <Button onClick={scrollTop} className='main-footer__btn-up' children='&#x2303;' />
            <div className='container'>
                <div className='main-footer__block-about'>
                    <div className='main-footer__about-us footer-about-us'>
                        <h4 className='footer-about-us__title bold'>Little about us</h4>
                        <p className='footer-about-us__latest-tweet'><span>Sed posuere</span> consectetur  est at. Nulla vitae elit libero, a pharetra.
                            Lorem ipsum <span>dolor sit</span> amet, conse ctetuer adipiscing elit.</p>
                    </div>
                    <div className='main-footer__socials footer-socials'>
                        <h4 className='footer-socials__title bold'>Socialize with us</h4>
                        <div className='footer-socials__socials social-icons'>
                            <a href='#' className='social-icons__icon social-icons__icon--fb'>
                                <i className='fab fa-facebook-f' />
                            </a>
                            <a href='#' className='social-icons__icon social-icons__icon--tw'>
                                <i className='fab fa-twitter' />
                            </a>
                            <a href='#' className='social-icons__icon social-icons__icon--drib'>
                                <i className='fab fa-dribbble' />
                            </a>
                            <a href='#' className='social-icons__icon social-icons__icon--digg'>
                                <i className='fab fa-digg' />
                            </a>
                            <a href='#' className='social-icons__icon social-icons__icon--mail'>
                                <i className='far fa-envelope' />
                            </a>
                        </div>
                    </div>
                </div>
                <div className='main-footer__archives footer-archives'>
                    <h4 className='footer-archives__title bold'>Our Archives</h4>
                    <div className='footer-archives__navbar archives-navbar'>
                        <a href='#' className='archives-navbar__link'>March 2012</a>
                        <a href='#' className='archives-navbar__link'>February 2012</a>
                        <a href='#' className='archives-navbar__link'>January 2012</a>
                        <a href='#' className='archives-navbar__link'>December 2011</a>
                    </div>
                </div>
                <div className='main-footer__posts footer-posts'>
                    <h4 className='footer-posts__title bold'>Popular Posts</h4>
                    <div className='footer-posts__posts-wrapper'>
                        <div className='footer-posts__post-block'>
                            <img src='./img/footer/post1.png' alt='album'/>
                            <div>
                                <h5 className='footer-posts__post-title'>Great Album</h5>
                                <p>
                                    <i className='fas fa-comment-alt' />
                                    <span>12 comments</span>
                                </p>
                            </div>
                        </div>
                        <div className='footer-posts__post-block'>
                            <img src='./img/footer/post2.png' alt='album'/>
                            <div>
                                <h5 className='footer-posts__post-title'>Great Album</h5>
                                <p>
                                    <i className='fas fa-comment-alt' />
                                    <span>12 comments</span>
                                </p>
                            </div>
                        </div>
                    </div>
                </div>
                <div className='main-footer__block-search'>
                    <div className='main-footer__search footer-search'>
                        <h4 className='footer-search__title bold'>Search our Site</h4>
                        <input type='text' name='search' placeholder='Enter Search ...'/>
                    </div>
                    <div className='main-footer__cloud footer-cloud'>
                        <h4 className='footer-cloud__title bold'>Tag Cloud</h4>
                        <div className='footer-cloud__links'>
                            <a href='#' className=''>Audio CD</a>
                            <a href='#'>Video</a>
                            <a href='#'>Playlist</a>
                            <a href='#'>Avantgarde</a>
                            <a href='#'>Melancholic</a>
                        </div>

                    </div>
                </div>
            </div>
        </div>
    );
}

export default Footer;