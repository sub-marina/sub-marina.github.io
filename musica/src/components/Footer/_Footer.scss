.main-footer {
  background: url(/img/footer-bg.png);
  color: #eaeaea;
  position: relative;

  &__upper-line {
    height: 6px;
    background-color: #000000;
  }

  .container {
    padding: 30px 0;
    @include flex(space-between);
    .bold {
      font-weight: bold;
    }
  }

  &__block-about {
    flex-basis: 212px;

    .footer-about-us {
      margin-bottom: 25px;

      &__latest-tweet {
        margin-top: 23px;
        font-size: 13px;
        line-height: 1.38;

        span {
          font-weight: bold;
        }
      }
    }

    .footer-socials {
      .social-icons {
        margin-top: 19px;

        &__icon {
          display: inline-block;
          margin-right: 3px;
          width: 23px;
          height: 23px;
          border-radius: 50%;
          color: white;
          font-size: 14px;
          text-align: center;
          line-height: 23px;
          background: rgba(255, 255, 255, .08);
        }
      }
    }
  }

  &__archives {
    flex-basis: 197px;

    .archives-navbar {
      margin-top: 15px;

      &__link {
        display: block;
        padding: 15px 0 13px;
        font-size: 13px;
        border-bottom: 1px dotted #eaeaea;

        &:last-child {
          border-bottom: none;
        }
      }
    }
  }

  .footer-posts {
    flex-basis: 212px;

    &__posts-wrapper {
      margin-top: 14px;
      display: flex;
      flex-direction: column;
    }

    &__post-block {
      @include flex(flex-start);

      img {
        margin-right: 10px;
        display: block;
      }

      p {
        font-size: 11px;
        text-transform: uppercase;
        i {
          margin-right: 7px;
        }
      }
    }

    &__post-title {
      margin-bottom: 13px;
      font-size: 13px;
      font-weight: bold;
    }
  }

  &__block-search {
    flex-basis: 214px;
  }
  .footer-search {
    margin-bottom: 31px;

    input {
      margin-top: 23px;
      width: 214px;
      padding: 13px 0;
      text-indent: 16px;
      box-shadow: 0 1px 0 0 rgba(0, 0, 0, 0.15);
      border: 1px solid #787979;
      background-color: rgba(0, 0, 0, 0.28);
      outline: none;

      &[placeholder] {
        font-size: 12px;
        font-style: italic;
        letter-spacing: -0.12px;
        color: #aaaaaa;
      }
    }
  }

  .footer-cloud {
    &__links {
      margin-top: 19px;

      a {
        display: inline-block;
        margin-right: 4px;
        margin-bottom: 4px;
        padding: 6px 7px;
        font-size: 12px;
        background-color: #2a2a2a;
      }
    }
  }
}