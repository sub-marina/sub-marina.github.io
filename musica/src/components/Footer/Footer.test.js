import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Footer from './Footer';

const mockScroll = jest.fn();
jest.mock('../Button/Button', () => () => (<button onClick={mockScroll}>Scroll</button>));

describe('Testing Footer.js', () => {
    test('Smoke test', () => {
        const { container } = render(<Footer />);
       console.log(container.innerHTML);
    });

    test('Footer snapshot', () => {
        const { container } = render(<Footer />);
        expect(container.innerHTML).toMatchSnapshot();
    });

    test('Click on button scroll page to top', () => {
        const { getByText } = render(<Footer />);

        fireEvent.click(getByText('Scroll'));
        expect(mockScroll).toHaveBeenCalledTimes(1);
    });
});