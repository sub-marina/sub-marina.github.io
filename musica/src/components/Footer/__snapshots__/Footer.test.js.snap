// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer.js Footer snapshot 1`] = `"<div class=\\"main-footer\\"><div class=\\"main-footer__upper-line\\"></div><button>Scroll</button><div class=\\"container\\"><div class=\\"main-footer__block-about\\"><div class=\\"main-footer__about-us footer-about-us\\"><h4 class=\\"footer-about-us__title bold\\">Little about us</h4><p class=\\"footer-about-us__latest-tweet\\"><span>Sed posuere</span> consectetur  est at. Nulla vitae elit libero, a pharetra. Lorem ipsum <span>dolor sit</span> amet, conse ctetuer adipiscing elit.</p></div><div class=\\"main-footer__socials footer-socials\\"><h4 class=\\"footer-socials__title bold\\">Socialize with us</h4><div class=\\"footer-socials__socials social-icons\\"><a href=\\"#\\" class=\\"social-icons__icon social-icons__icon--fb\\"><i class=\\"fab fa-facebook-f\\"></i></a><a href=\\"#\\" class=\\"social-icons__icon social-icons__icon--tw\\"><i class=\\"fab fa-twitter\\"></i></a><a href=\\"#\\" class=\\"social-icons__icon social-icons__icon--drib\\"><i class=\\"fab fa-dribbble\\"></i></a><a href=\\"#\\" class=\\"social-icons__icon social-icons__icon--digg\\"><i class=\\"fab fa-digg\\"></i></a><a href=\\"#\\" class=\\"social-icons__icon social-icons__icon--mail\\"><i class=\\"far fa-envelope\\"></i></a></div></div></div><div class=\\"main-footer__archives footer-archives\\"><h4 class=\\"footer-archives__title bold\\">Our Archives</h4><div class=\\"footer-archives__navbar archives-navbar\\"><a href=\\"#\\" class=\\"archives-navbar__link\\">March 2012</a><a href=\\"#\\" class=\\"archives-navbar__link\\">February 2012</a><a href=\\"#\\" class=\\"archives-navbar__link\\">January 2012</a><a href=\\"#\\" class=\\"archives-navbar__link\\">December 2011</a></div></div><div class=\\"main-footer__posts footer-posts\\"><h4 class=\\"footer-posts__title bold\\">Popular Posts</h4><div class=\\"footer-posts__posts-wrapper\\"><div class=\\"footer-posts__post-block\\"><img src=\\"./img/footer/post1.png\\" alt=\\"album\\"><div><h5 class=\\"footer-posts__post-title\\">Great Album</h5><p><i class=\\"fas fa-comment-alt\\"></i><span>12 comments</span></p></div></div><div class=\\"footer-posts__post-block\\"><img src=\\"./img/footer/post2.png\\" alt=\\"album\\"><div><h5 class=\\"footer-posts__post-title\\">Great Album</h5><p><i class=\\"fas fa-comment-alt\\"></i><span>12 comments</span></p></div></div></div></div><div class=\\"main-footer__block-search\\"><div class=\\"main-footer__search footer-search\\"><h4 class=\\"footer-search__title bold\\">Search our Site</h4><input type=\\"text\\" name=\\"search\\" placeholder=\\"Enter Search ...\\"></div><div class=\\"main-footer__cloud footer-cloud\\"><h4 class=\\"footer-cloud__title bold\\">Tag Cloud</h4><div class=\\"footer-cloud__links\\"><a href=\\"#\\" class=\\"\\">Audio CD</a><a href=\\"#\\">Video</a><a href=\\"#\\">Playlist</a><a href=\\"#\\">Avantgarde</a><a href=\\"#\\">Melancholic</a></div></div></div></div></div>"`;
